GVFS/GVFS.Platform.Mac/MacFileBasedLock.cs

```diff
@@ -0,0 +1,159 @@
+﻿using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace GVFS.Platform.Mac
+{
+    public class MacFileBasedLock : FileBasedLock
+    {
+        // #define O_WRONLY    0x0001      /* open for writing only */
+        private const int OpenWriteOnly = 0x0001; 

```

Comment:
Nit: since you created a `NativeMethods` subclass (which I like), I would also move all of these constants inside there since that is the only context in which the constants are needed

Reply:
Thats a good idea, I'll move these